### PR TITLE
Update test-e2e script

### PR DESCRIPTION
**Pull request by miry**
Extract execution of benchmark script to function.
Use alias for command lines.
Extract trap eval string to function.

### CODE REVIEW

File: scripts/test-e2e [hunk 1]
```diff
@@ -2,36 +2,58 @@
...
+cd $(dirname "$0")
+
+server="../dist/toxiproxy-server"
+
+benchmark() {
+  go test -bench=.. ../test/e2e -v
+}
+
+cli() {
+  ../dist/toxiproxy-cli $* 2>&1 | sed -e 's/^/[client] /'
```
Comment (barthalion) [inline scripts/test-e2e:14]:
```suggestion
  ../dist/toxiproxy-cli "$@" 2>&1 | sed -e 's/^/[client] /'
```

This is to ensure expanded argv is still quoted if arguments include special characters.

File: scripts/test-e2e [hunk 2]
```diff
@@ -2,36 +2,58 @@
...
 wait_for_url() {
   curl -s --retry-connrefused --retry 5 --retry-delay 2 --retry-max-time 30 \
        --max-time 1 -L -I -X GET "${1}"
 }
 
 # Stop all background jobs on exit
-trap 'pkill -15 -f "dist/toxiproxy-server -proxy-metrics -runtime-metrics$"; pkill -15 -f "exe/endpoint$"' \
-     EXIT SIGINT SIGTERM
+function cleanup() {
+  sig="$1"
```
Comment (barthalion) [inline scripts/test-e2e:24]:
```suggestion
  local sig="$1"
```

Bash variables are global even if declared in a function.

**Review by barthalion — COMMENTED** [review]
Just some nitpicks, feel free to ignore if they don't make sense in wider context.

File: scripts/test-e2e (hunk 2 above)
Comment (miry) [reply scripts/test-e2e]: nice. fixed.

File: scripts/test-e2e (hunk 1 above)
Comment (miry) [reply scripts/test-e2e]: thank you, fixed.

**Comment by miry** [comment]
merged in commit 29685f4cd84df09161da9a58674484e9a83fcef2

**Pull request closed without merging**